net/dns/resolver/forwarder.go

```diff
@@ -59,9 +60,12 @@ func truncatedFlagSet(pkt []byte) bool {
 
 const (
 	// dohTransportTimeout is how long to keep idle HTTP
-	// connections open to DNS-over-HTTPs servers. This is pretty
-	// arbitrary.
-	dohTransportTimeout = 30 * time.Second
+	// connections open to DNS-over-HTTPs servers. 5 seconds is a
+	// sensible default, as it's long enough to handle a burst of
+	// queries coming in a row, but short enough to not keep idle
+	// connections open for too long (which affects battery life on
+	// mobile devices).
+	dohTransportTimeout = 5 * time.Second
```

Comment:
lets rename this to `dohIdleConnTimeout`, I appreciate the extra comments, but the current name is still vague.

A shorter timeout sounds good to me for now, I'd love for us to make this more adaptive in the future, shutting down conns on mobile when battery is idle, but keeping them up while in a wake state, etc.

On the specific value for a better idle timeout, ideally we should keep this just below common TCPKeepAlive timers. We could do a quick sample of what those are for DoH servers we currently operate against. Shutting down before entering keepalive should avoid some worst-case extra RTO conditions and so on.

Reply:
Renamed; set timeout to 10 seconds as per Slack discussion.

Reply:
Oh, now I see this "just below common TCPKeepAlive timers". That's worth putting in a comment as a justification for the value.